ossl/src/lib.rs

```diff
@@ -704,6 +710,17 @@ impl<'a> OsslParam<'a> {
         }))
     }
 
+    /// Creates an `OsslParam` from an OpenSSl provider pointer that contains
+    /// allocated data that can be safely zeroized before freeing the pointer
+    /// Use with care as it will overwrite any data pointed by the params
+    pub fn from_allocated_params_ptr(
```

Comment:
actually considering if the function name should be more self-explanatory like `from_sensitive_params_ptr()` or something along these lines. The "allocated" does not really say the casual reader that it should zeroize the buffers.

Reply:
Looks good!